api/src/main/java/org/openmrs/module/clientregistry/api/impl/FhirCRPatientServiceImpl.java

```diff
@@ -61,9 +61,8 @@ public IBundleProvider getPatientsByPIX(String sourceIdentifier, String sourceId
 		}
 		
 		Parameters crMatchingParams = identifiersRequest.useHttpGet().execute();
-		List<String> crIdentifiers = crMatchingParams.getParameter().stream()
-		        .filter(param -> Objects.equals(param.getName(), "targetId")).map(param -> param.getValue().toString())
-		        .collect(Collectors.toList());
+		List<String> crIdentifiers = crMatchingParams.getParameter().stream().filter(param -> Objects.equals(param.getName(), "targetId"))
+				.map(param -> ((Reference) param.getValue()).getReference()).collect(Collectors.toList());
```

Comment:
This change is mainly because of https://profiles.ihe.net/ITI/PIXm/Parameters-pixm-response-mohralice-red-all.json.html which doesn't seem to have been correctly implemented in opencr hence this.